nim-compile.el

```diff
@@ -126,28 +105,26 @@ The config file would one of those: config.nims, PROJECT.nim.cfg, or nim.cfg."
    (delq nil `(,nim-compile-command ,@args ,@nim-compile-user-args ,file))
    " "))
 
-(define-derived-mode nim-compile-mode compilation-mode "nim-compile"
-  "major-mode for nim compilation buffer.")
-
 ;;;###autoload
 (defun nim-compile ()
   "Compile and execute the current buffer as a nim file.  All output is writton into the *compilation* buffer."
   (interactive)
   (when (derived-mode-p 'nim-mode)
-    (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
-    (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
-    (nim-compile--set-compile-command)
-    (funcall 'compile compile-command 'nim-compile-mode)))
+    ;; (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
+    ;; (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
+    ;; (nim-compile--set-compile-command)
+    ;; (funcall 'compile compile-command 'nim-compile-mode)))
+    (call-interactively 'compile)))
 
 (require 'ansi-color)
 (defun nim--colorize-compilation-buffer ()
-  "Colorize compilation buffer."
+  "Colorize compilation buffer.  Add to `compilation-filter-hook', when you want to interpret terminal colors in output buffer."
   (let ((inhibit-read-only t))
     (ansi-color-apply-on-region compilation-filter-start (point-max))))
 
 (defun nim--remove-colorize-hook (_buf _process-state)
-  "Remove ‘nim--colorize-compilation-buffer’."
-  (when (get-buffer "*nim-compile*")
+  "Remove ‘nim--colorize-compilation-buffer’.  Currently unused."
+  (when (get-buffer "*compilation*")
```

Comment:
Okay. Regarding name of the `*compilation*` buffer, I wouldn't use it for some reasons:

1. if you share emacs (using emacsclient) with other people, you may accidentally contaminate other people's `*compilation*` buffer (it does even if you make other name, but at least I want to avoid using default's `*compilation*` buffer)
2. on 1's situation (I mean using emacsclient with multiple users), if someone customized something for `*compilation*` buffer, it hard to fix from other users.
3. just my personal preference; I saw several programming major-modes, and I think it's majority that specify major-mode specific buffer.
4. since this is Nim dedicated package, why you need abstract `*compilation*` buffer? I would remain that for personal use.